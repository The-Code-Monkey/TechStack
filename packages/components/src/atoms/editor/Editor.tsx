import EditorContainer from "@techstack/lexical";

import "@techstack/lexical/css";

export const Editor = ({ name }) => {
    return (
        <div>
            <EditorContainer name={name} value={''} onChange={() => {}} />
            <EditorContainer name={name} value={'<p>test</p>'}  onChange={() => {}} />
            <EditorContainer name={name} value={''}  onChange={() => {}} />
        </div>
    )
}

export default Editor;