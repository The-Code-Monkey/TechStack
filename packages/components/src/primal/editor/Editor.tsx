import ReactQuill from '@techstack/react-quill';

import '@techstack/react-quill/dist/quill.snow.css';

export const Editor = () => {
  return (
    <div>
      <ReactQuill />
    </div>
  );
};

export default Editor;
