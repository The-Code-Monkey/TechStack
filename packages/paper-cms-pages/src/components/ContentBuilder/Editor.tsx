import { Box } from '@techstack/components';
import ReactQuill from 'react-quill';
import 'react-quill/dist/quill.core.css';
import 'react-quill/dist/quill.snow.css';

import { EditorWrapper, StyledAccordion } from './styled';

interface Props {
  value: string;
  onChange: (e: any) => void;
  name?: string;
}

const Editor = ({ value, onChange, name }: Props) => {
  const handleOnChange = (value?: string) => {
    onChange({
      target: {
        name,
        value,
      },
    });
  };

  return (
    <Box d='flex' flexDir='column' w='full'>
      <StyledAccordion
        title={
          value
            ?.split('</h1>')[0]
            .replace(/(<[a-zA-Z0-9]>|<\/[a-zA-Z0-9]>)/g, '')
            .replace('&amp;', '&') ?? 'No Content'
        }
        maxHeight={7000}
      >
        <EditorWrapper>
          <ReactQuill theme={'snow'} value={value} onChange={handleOnChange} />
        </EditorWrapper>
      </StyledAccordion>
    </Box>
  );
};

export default Editor;
