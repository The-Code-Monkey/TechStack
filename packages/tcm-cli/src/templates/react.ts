import { PackageJson } from 'type-fest';

import basicTemplate from './basic';
import { Template } from './template';

const reactTemplate: Template = {
  name: 'react',
  dependencies: [
    ...basicTemplate.dependencies,
    '@types/react',
    '@types/react-dom',
    'react',
    'react-dom',
    '@tsconfig/create-react-app',
  ],
  packageJson: {
    ...basicTemplate.packageJson,
    peerDependencies: {
      react: '>=16',
    },
    scripts: {
      ...basicTemplate.packageJson.scripts,
      test: 'tcm test --passWithNoTests',
    } as PackageJson['scripts'],
    jest: {
      testEnvironment: 'jsdom',
    },
  },
};

export default reactTemplate;
