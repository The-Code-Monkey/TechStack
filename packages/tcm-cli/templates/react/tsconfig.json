{
  // see https://www.typescriptlang.org/tsconfig to better understand tsconfigs
  "extends": "../../tsconfig.json",
  "include": ["src", "types"],
"exclude": ["src/**/__test__/*.test.ts*x"],
}
