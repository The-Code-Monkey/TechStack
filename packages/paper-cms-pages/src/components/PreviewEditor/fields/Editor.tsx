import { useCallback, useState } from 'react';
import ReactQuill from 'react-quill';
import 'react-quill/dist/quill.core.css';

import { DefaultRecordType } from '../../../utils/types';
import { EditorWrapper } from '../styled';

interface Props {
  item: DefaultRecordType;
  styles: Record<string, string>;
  onChange: (index: string, value: string) => void;
  onHoverChange: (hoveredElement: string, isTop: boolean) => void;
  hoveredElement: [string, boolean];
  onReorder: (thisId: string, thatId: string, top: boolean) => void;
}
const Editor = ({
  item,
  styles,
  onChange,
  onHoverChange,
  hoveredElement: [hoveredId, isTop],
  onReorder,
}: Props) => {
  const [isDragging, setIsDragging] = useState(false);

  const handleOnChange = (index: string) => (value: string) => {
    onChange(index, value);
  };

  const findCorrectTarget = useCallback(
    (parentTarget: any): any => {
      return parentTarget.className.includes(item.id)
        ? parentTarget
        : findCorrectTarget(parentTarget.parentElement);
    },
    [item.id]
  );

  const handleOnDragOver = useCallback(
    (e: any) => {
      if (!isDragging) {
        e.preventDefault();
        let target = e.target;
        if (!target.className.includes(item.id)) {
          target = findCorrectTarget(target.parentElement);
        }

        const box = target.getBoundingClientRect();
        const offsetTop = e.clientY - box.top - box.height / 2;

        const id = target.className
          .split(' ')
          .find((str: string) => str.startsWith('#'))
          .split('_')[1];

        onHoverChange(id, offsetTop < 0);
      }
    },
    [findCorrectTarget, isDragging, item.id, onHoverChange]
  );

  const handleOnDragEnd = () => {
    setIsDragging(false);
    onReorder(item.id, hoveredId, isTop);
  };

  const className = `innerSection_${item.id}`;

  const isHovered = hoveredId === item.id;

  return (
    <EditorWrapper
      {...(styles ?? {})}
      draggable
      onDragOver={handleOnDragOver}
      onDragStart={() => setIsDragging(true)}
      onDragEnd={handleOnDragEnd}
      className={`#${className}`}
      isTop={isTop}
      isHovered={isHovered}
    >
      <ReactQuill
        theme={false as any}
        value={item.value}
        onChange={handleOnChange(item.id)}
      />
    </EditorWrapper>
  );
};

export default Editor;
