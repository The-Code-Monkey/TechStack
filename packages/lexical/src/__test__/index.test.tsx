import Editor from '../';
import { render } from '@testing-library/react';

describe('<Editor />', () => {
    it('renders text correctly', () => {
        const {asFragment} = render(<Editor name={'test'} value={''} onChange={jest.fn()}/>);

        expect(asFragment()).toMatchSnapshot();
    })
})