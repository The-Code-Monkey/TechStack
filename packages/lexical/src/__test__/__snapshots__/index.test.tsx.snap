// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Editor /> renders text correctly 1`] = `
<DocumentFragment>
  <div
    class="editor-container"
  >
    <div
      class="toolbar"
    >
      <button
        aria-label="Formatting Options"
        class="toolbar-item block-controls"
        type="button"
      >
        <span
          class="icon block-type paragraph"
        >
          <svg
            fill="currentColor"
            height="14"
            stroke="currentColor"
            viewBox="0 0 16 16"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 12.5a.5.5 0 0 1 .5-.5h7a.5.5 0 0 1 0 1h-7a.5.5 0 0 1-.5-.5zm0-3a.5.5 0 0 1 .5-.5h11a.5.5 0 0 1 0 1h-11a.5.5 0 0 1-.5-.5zm0-3a.5.5 0 0 1 .5-.5h11a.5.5 0 0 1 0 1h-11a.5.5 0 0 1-.5-.5zm4-3a.5.5 0 0 1 .5-.5h7a.5.5 0 0 1 0 1h-7a.5.5 0 0 1-.5-.5z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="text"
        >
          Normal
        </span>
        <i
          class="chevron-down"
        >
          <svg
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="6 9 12 15 18 9"
            />
          </svg>
        </i>
      </button>
      <div
        class="divider"
      />
      <button
        aria-label="Format Bold"
        class="toolbar-item spaced "
        type="button"
      >
        <i
          class="format bold"
        >
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 4h8a4 4 0 0 1 4 4 4 4 0 0 1-4 4H6z"
            />
            <path
              d="M6 12h9a4 4 0 0 1 4 4 4 4 0 0 1-4 4H6z"
            />
          </svg>
        </i>
      </button>
      <button
        aria-label="Format Italics"
        class="toolbar-item spaced "
        type="button"
      >
        <i
          class="format italic"
        >
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="19"
              x2="10"
              y1="4"
              y2="4"
            />
            <line
              x1="14"
              x2="5"
              y1="20"
              y2="20"
            />
            <line
              x1="15"
              x2="9"
              y1="4"
              y2="20"
            />
          </svg>
        </i>
      </button>
      <button
        aria-label="Format Underline"
        class="toolbar-item spaced "
        type="button"
      >
        <i
          class="format underline"
        >
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 3v7a6 6 0 0 0 6 6 6 6 0 0 0 6-6V3"
            />
            <line
              x1="4"
              x2="20"
              y1="21"
              y2="21"
            />
          </svg>
        </i>
      </button>
      <button
        aria-label="Format Strikethrough"
        class="toolbar-item spaced "
        type="button"
      >
        <i
          class="format strikethrough"
        >
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 16 16"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.333 5.686c0 .31.083.581.27.814H5.166a2.776 2.776 0 0 1-.099-.76c0-1.627 1.436-2.768 3.48-2.768 1.969 0 3.39 1.175 3.445 2.85h-1.23c-.11-1.08-.964-1.743-2.25-1.743-1.23 0-2.18.602-2.18 1.607zm2.194 7.478c-2.153 0-3.589-1.107-3.705-2.81h1.23c.144 1.06 1.129 1.703 2.544 1.703 1.34 0 2.31-.705 2.31-1.675 0-.827-.547-1.374-1.914-1.675L8.046 8.5H1v-1h14v1h-3.504c.468.437.675.994.675 1.697 0 1.826-1.436 2.967-3.644 2.967z"
            />
          </svg>
        </i>
      </button>
      <button
        aria-label="Insert Code"
        class="toolbar-item spaced "
        type="button"
      >
        <i
          class="format code"
        >
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="16 18 22 12 16 6"
            />
            <polyline
              points="8 6 2 12 8 18"
            />
          </svg>
        </i>
      </button>
      <button
        aria-label="Insert Link"
        class="toolbar-item spaced "
        type="button"
      >
        <i
          class="format link"
        >
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 13a5 5 0 0 0 7.54.54l3-3a5 5 0 0 0-7.07-7.07l-1.72 1.71"
            />
            <path
              d="M14 11a5 5 0 0 0-7.54-.54l-3 3a5 5 0 0 0 7.07 7.07l1.71-1.71"
            />
          </svg>
        </i>
      </button>
      <button
        aria-label="Formatting text color"
        class="toolbar-item color-picker"
        type="button"
      >
        <span
          class="format font-color"
        >
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 512 512"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M221.631 109 109.92 392h58.055l24.079-61h127.892l24.079 61h58.055L290.369 109Zm-8.261 168L256 169l42.63 108Z"
              fill="#777"
            />
          </svg>
        </span>
        <i
          class="chevron-down"
        >
          <svg
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="6 9 12 15 18 9"
            />
          </svg>
        </i>
      </button>
      <button
        aria-label="Formatting background color"
        class="toolbar-item color-picker"
        type="button"
      >
        <span
          class="format bg-color"
        >
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 48 48"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h48v48H0z"
              fill="#fff"
              fill-opacity=".01"
            />
            <path
              clip-rule="evenodd"
              d="M37 37a4 4 0 0 0 4-4c0-1.473-1.333-3.473-4-6-2.667 2.527-4 4.527-4 6a4 4 0 0 0 4 4Z"
              fill="#777"
              fill-rule="evenodd"
            />
            <path
              d="m20.854 5.504 3.535 3.536"
              stroke="#777"
              stroke-linecap="round"
              stroke-width="4"
            />
            <path
              d="M23.682 8.333 8.125 23.889 19.44 35.203l15.556-15.557L23.682 8.333Z"
              stroke="#777"
              stroke-linejoin="round"
              stroke-width="4"
            />
            <path
              d="m12 20.073 16.961 5.577M4 43h40"
              stroke="#777"
              stroke-linecap="round"
              stroke-width="4"
            />
          </svg>
        </span>
        <i
          class="chevron-down"
        >
          <svg
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="6 9 12 15 18 9"
            />
          </svg>
        </i>
      </button>
      <div
        class="divider"
      />
      <button
        type="button"
      >
        <i
          class="format element-format"
        >
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="5"
              y2="19"
            />
            <line
              x1="5"
              x2="19"
              y1="12"
              y2="12"
            />
          </svg>
        </i>
         Table
      </button>
      <button
        aria-label="Left Align"
        class="toolbar-item spaced"
        type="button"
      >
        <i
          class="format left-align"
        >
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="17"
              x2="3"
              y1="10"
              y2="10"
            />
            <line
              x1="21"
              x2="3"
              y1="6"
              y2="6"
            />
            <line
              x1="21"
              x2="3"
              y1="14"
              y2="14"
            />
            <line
              x1="17"
              x2="3"
              y1="18"
              y2="18"
            />
          </svg>
        </i>
      </button>
      <button
        aria-label="Center Align"
        class="toolbar-item spaced"
        type="button"
      >
        <i
          class="format center-align"
        >
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="18"
              x2="6"
              y1="10"
              y2="10"
            />
            <line
              x1="21"
              x2="3"
              y1="6"
              y2="6"
            />
            <line
              x1="21"
              x2="3"
              y1="14"
              y2="14"
            />
            <line
              x1="18"
              x2="6"
              y1="18"
              y2="18"
            />
          </svg>
        </i>
      </button>
      <button
        aria-label="Right Align"
        class="toolbar-item spaced"
        type="button"
      >
        <i
          class="format right-align"
        >
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="21"
              x2="7"
              y1="10"
              y2="10"
            />
            <line
              x1="21"
              x2="3"
              y1="6"
              y2="6"
            />
            <line
              x1="21"
              x2="3"
              y1="14"
              y2="14"
            />
            <line
              x1="21"
              x2="7"
              y1="18"
              y2="18"
            />
          </svg>
        </i>
      </button>
      <button
        aria-label="Justify Align"
        class="toolbar-item"
        type="button"
      >
        <i
          class="format justify-align"
        >
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="21"
              x2="3"
              y1="10"
              y2="10"
            />
            <line
              x1="21"
              x2="3"
              y1="6"
              y2="6"
            />
            <line
              x1="21"
              x2="3"
              y1="14"
              y2="14"
            />
            <line
              x1="21"
              x2="3"
              y1="18"
              y2="18"
            />
          </svg>
        </i>
      </button>
       
    </div>
    <div
      class="editor-inner"
    >
      <div
        class="editor"
      >
        <div
          class="editor-input"
          contenteditable="true"
          data-lexical-editor="true"
          role="textbox"
          spellcheck="true"
          style="user-select: text; white-space: pre-wrap; word-break: break-word;"
        />
      </div>
      <div
        class="editor-placeholder"
      >
        Enter some rich text...
      </div>
    </div>
  </div>
</DocumentFragment>
`;
